capa/main.py

```diff
@@ -978,6 +978,41 @@ def handle_common_args(args):
         args.signatures = sigs_path
 
 
+def last_resort_exception_handler(args, exctype=None, value=None, traceback=None):
+    """
+    custom exception handler to replace the default sys.excepthook,
+    prints traceback in debug mode, otherwise prints helpful message
+
+    using "nonlocal" keyword and closure code construct enables
+    last_resort_exception_handler args to first be initialized and
+    then be updated if the function needs to handle an exception
+
+    args:
+      args (argparse.Namespace): parsed command-line arguments
+      exctype (Exception): exception class
+      value (str): exception instance
+      traceback (TracebackType): a traceback object
+    """
+
+    def handle_exception_information(args):
+        try:
+            nonlocal exctype, value, traceback
+            if "-d" or "--debug" in args:
+                return sys.__excepthook__(exctype, value, traceback)
+            else:
+                print(
+                    f"Unexpected exception raised: {exctype}. Please run capa in debug mode (-d/--debug) "
+                    + "to see the stack trace. Please also report your issue on the capa GitHub page so we "
+                    + "can improve the code! (https://github.com/mandiant/capa/issues)"
+                )
+
+        except Exception as e:  # pylint: disable=broad-except
+            logger.error("%s", str(e))
+            return  # should this return an object like some other capa try-except statements, e.g., E_CORRUPT_FILE or E_INVALID_FILE_TYPE?
```

Comment:
yes, if this returns the value that is then used as program return code

since it's not known an error name that indicates this would be good, e.g. `E_UNHANDLED`?